### PR TITLE
add js document processor

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: plugins/Microsoft.DocAsCode.Build.JavaScriptReference/JavaScriptReferenceDocumentProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                                return ProcessingPriority.NotSupported;
+                        }
+                    }
+
+                    if (".jsyml".Equals(Path.GetExtension(file.File), StringComparison.OrdinalIgnoreCase) ||
+                        ".jsyaml".Equals(Path.GetExtension(file.File), StringComparison.OrdinalIgnoreCase))
+                    {
+                        return ProcessingPriority.Normal;
+                    }
+
```
Comment (vwxyzh) [inline plugins/Microsoft.DocAsCode.Build.JavaScriptReference/JavaScriptReferenceDocumentProcessor.cs]: remove

File: plugins/Microsoft.DocAsCode.Build.JavaScriptReference/JavaScriptReferenceDocumentProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                    {
+                        var mime = YamlMime.ReadMime(file.File);
+                        switch (mime)
+                        {
+                            case YamlMime.JavaScriptReference:
+                                return ProcessingPriority.Normal;
+                            case null:
+                                // A YAML without YamlMime is treated as ManagedReference by default
+                                return ProcessingPriority.NotSupported;
+                            default:
```
Comment (vwxyzh) [inline plugins/Microsoft.DocAsCode.Build.JavaScriptReference/JavaScriptReferenceDocumentProcessor.cs]: remove

File: src/Microsoft.DocAsCode.Common/YamlMime.cs [hunk 3]
```diff
@@ -14,6 +14,7 @@ public static class YamlMime
         public const string ManagedReference = YamlMimePrefix + nameof(ManagedReference);
         public const string TableOfContent = YamlMimePrefix + nameof(TableOfContent);
         public const string XRefMap = YamlMimePrefix + nameof(XRefMap);
+        public const string JavaScriptReference = YamlMimePrefix + nameof(JavaScriptReference);
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: move to JavaScriptReferenceDocumentProcessor.cs

File: plugins/Microsoft.DocAsCode.Build.JavaScriptReference/JavaScriptReferenceDocumentProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    using Microsoft.DocAsCode.Plugins;
+
+    using Newtonsoft.Json;
+
+    [Export(typeof(IDocumentProcessor))]
+    public class JavaScriptReferenceDocumentProcessor : ReferenceDocumentProcessorBase
+    {
+        #region ReferenceDocumentProcessorBase Members
+
+        protected override string ProcessedDocumentType { get; } = "JavaScriptReference";
```
Comment (vwxyzh) [inline plugins/Microsoft.DocAsCode.Build.JavaScriptReference/JavaScriptReferenceDocumentProcessor.cs]: >{ get; } = [](start = 56, length = 10)

`=>`

**Comment by vwxyzh** [comment]
:shipit:

File: plugins/Microsoft.DocAsCode.Build.JavaScriptReference/ApplyOverwriteDocumentForJSREF.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    using System.Collections.Generic;
+    using System.Composition;
+    using System.Linq;
+
+    using Microsoft.DocAsCode.Build.Common;
+    using Microsoft.DocAsCode.DataContracts.Common;
+    using Microsoft.DocAsCode.Plugins;
+
+    [Export(nameof(JavaScriptReferenceDocumentProcessor), typeof(IDocumentBuildStep))]
+    public class ApplyOverwriteDocumentForJsRef : ApplyOverwriteDocument
```
Comment (vicancy) [inline plugins/Microsoft.DocAsCode.Build.JavaScriptReference/ApplyOverwriteDocumentForJSREF.cs]: >Js [](start = 42, length = 2)

actually it is JS.....

File: plugins/Microsoft.DocAsCode.Build.JavaScriptReference/Constants.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.JavaScriptReference
+{
+    using Microsoft.DocAsCode.Common;
+
+    internal static class Constants
+    {
+        public const string JavaScriptReferenceYamlMime = YamlMime.YamlMimePrefix + "JavaScriptReference";
```
Comment (vicancy) [inline plugins/Microsoft.DocAsCode.Build.JavaScriptReference/Constants.cs]: >"JavaScriptReference [](start = 84, length = 20)

also constant?

**Comment by vicancy** [comment]
:shipit:

**Pull request merged**